File: petcarehub/src/main/java/com/petcarehub/cart/service/CartService.java
package com.petcarehub.cart.service;

import com.petcarehub.cart.dto.CartResponseDto;

public interface CartService {
    CartResponseDto getCart(Long userId);
    CartResponseDto addItem(Long userId, Long productId, Integer quantity);
    CartResponseDto updateQuantity(Long userId, Long productId, Integer quantity);
    CartResponseDto removeItem(Long userId, Long productId);
}

File: frontend/src/features/cart/services/cartService.js
import { httpClient } from "./httpClient";

export const cartService = {
  getCart: async (userId) => (await httpClient.get(`/api/cart/${userId}`)).data,
  addItem: async (userId, productId, quantity = 1) =>
    (await httpClient.post(`/api/cart/${userId}/items`, { productId, quantity })).data,
  updateQuantity: async (userId, productId, quantity) =>
    (await httpClient.put(`/api/cart/${userId}/items/${productId}`, { quantity })).data,
  removeItem: async (userId, productId) =>
    (await httpClient.delete(`/api/cart/${userId}/items/${productId}`)).data
};
